rct229/rulesets/ashrae9012019/ruleset_functions/get_zone_BPF_BAT_test.py

```diff
@@ -0,0 +1,105 @@
+                        {
+                            "id": "zone_1",
+                            "spaces": [
+                                # sll the lighting_space_type fall into `UNDETERMINED`
+                                {
+                                    "id": "Space 1_1",
+                                    "lighting_space_type": "ATRIUM_LOW_MEDIUM",
+                                    "floor_area": 100,
+                                },
+                                {
+                                    "id": "Space 1_2",
+                                    "lighting_space_type": "CLASSROOM_LECTURE_HALL_TRAINING_ROOM_ALL_OTHER",
+                                    "floor_area": 200,
+                                },
+                            ],
+                        },
+                        {
+                            "id": "zone_2",
+                            "spaces": [
+                                # lighting_space_type fallS into `HEALTHCARE_HOSPITAL` AND `ALL_OTHER`
+                                {
+                                    "id": "Space 2_1",
+                                    "lighting_space_type": "HEALTHCARE_FACILITY_NURSES_STATION",
+                                    "floor_area": 300,
+                                },
+                                {
+                                    "id": "Space 2_2",
+                                    "lighting_space_type": "HEALTHCARE_FACILITY_PATIENT_ROOM",
+                                    "floor_area": 400,
+                                },
+                                {
+                                    "id": "Space 2_3",
+                                    "lighting_space_type": "MANUFACTURING_FACILITY_DETAILED_MANUFACTURING_AREA",
+                                    "floor_area": 100,
+                                },
+                            ],
+                        },
+                        {
+                            "id": "zone_3",
+                            "spaces": [
+                                # to test missing lighting_space_type key
+                                {
+                                    "id": "Space 3_1",
+                                }
+                            ],
+                        },
+                    ],
+                }
+            ],
+        }
+    ],
+    "type": "BASELINE_0",
+}
+
+
+TEST_RPD = {
+    "id": "ASHRAE229",
+    "ruleset_model_descriptions": [TEST_BUILDING],
+    "data_timestamp": "2024-02-12T09:00Z",
+}
+
+TEST_RMD = quantify_rmr(TEST_RPD)["ruleset_model_descriptions"][0]
+
+
+def test__TEST_RMD__is_valid():
+    schema_validation_result = schema_validate_rmr(TEST_RPD)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__get_zone_BPF_BAT__all_undetermined():
+    assert get_zone_BPF_BAT(TEST_RMD, "zone_1") == {"UNDETERMINED": 300 * ureg("m2")}
```

Comment:
Should add one more test, a zone has no spaces scenario.

Reply:
Addressed.